app/controllers/katello/api/v2/systems_controller.rb

```diff
@@ -383,8 +383,7 @@ def system_params(param_hash)
                                                      :guest_ids, :host_collection_ids => [])
 
       system_params[:facts] = param_hash[:system][:facts].permit! if param_hash[:system][:facts]
-      system_params[:cp_type] = param_hash[:type] ? param_hash[:type] : ::Katello::System::DEFAULT_CP_TYPE
-      system_params.delete(:type) if param_hash[:system].key?(:type)
+      system_params[:type] = param_hash[:type] ? param_hash[:type] : ::Katello::System::DEFAULT_CP_TYPE
```

Comment:
Can you move  ::Katello::System::DEFAULT_CP_TYPE to the subscription aspect?  or the maybe even the  candlepin consumer service?